echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/controller/NotificationController.groovy

```diff
@@ -32,8 +38,25 @@ class NotificationController {
   @Autowired(required=false)
   Collection<NotificationService> notificationServices
 
+  @Autowired
+  InteractiveNotificationCallbackHandler callbackHandler
+
   @RequestMapping(method = RequestMethod.POST)
   EchoResponse create(@RequestBody Notification notification) {
     notificationServices?.find { it.supportsType(notification.notificationType) }?.handle(notification)
   }
+
+  @RequestMapping(
+    value = '/callbacks/{source}',
```

Comment:
This would be a new external, unauthenticated API exposed through gate.

Reply:
I think Slack will include a token/shared-secret in this request that can be used to verify it was the sender - can checking that be part of this flow?

Reply:
Yes, absolutely! I already had a `TODO` somewhere to do that. What I meant here was just that, unlike other calls coming through gate, this won’t be using a Spinnaker identity to authenticate.